pkg/placement/ocm.go

```diff
@@ -173,7 +173,22 @@ func (op *ocmPlacer) Place(ctx context.Context, upStreamGateway *gatewayv1beta1.
 			return existingClusters, err
 		}
 
-		if err := gracePeriod.GracefulDelete(ctx, op.c, w); err != nil {
+		// Check if the ManagedCluster still exists,
+		// otherwise delete without any grace period.
+		// This can happen if a ManagedCluster is deleted,
+		// as opposed to just a placement decision changing
+		ignoreGrace := false
+		managedCluster := &clusterv1.ManagedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cluster,
+			},
+		}
+		err = op.c.Get(ctx, client.ObjectKeyFromObject(managedCluster), managedCluster)
+		if k8serrors.IsNotFound(err) {
+			log.V(3).Info(fmt.Sprintf("ManagedCluster not found '%s', ignoring grace period", cluster))
+			ignoreGrace = true
+		}
+		if err := gracePeriod.GracefulDelete(ctx, op.c, w, ignoreGrace); err != nil {
```

Comment:
In order for  this to get here on cluster deletion, it must be updating the PlacementDecision as expected and the cluster that was deleted removed?

Reply:
nvm, this is answered here https://github.com/Kuadrant/multicluster-gateway-controller/issues/303#issuecomment-1635890565